application.py

```diff
@@ -102,9 +94,57 @@ def userhome(user):
 
 @app.route("/logout/<username>")
 def logout(username):
+    print("Entered")
     session.pop(username, None)
     return redirect(url_for('index'))
 
+@app.route("/bookpage",methods=["POST","GET"])
+def bookspage():
+    print("Entered in b")
+    for key in session.keys():
+        username = key
+    book = bookreview("1416949658", "The Dark Is Rising", "Susan Cooper", 1973)
+    res = requests.get("https://www.goodreads.com/book/review_counts.json",
+                       params={"key": "2VIV9mRWiAq0OuKcOPiA", "isbns": book.isbn})
+    data = res.text
+    parsed = json.loads(data)
+    print(parsed)
+    res = {}
+    for i in parsed:
+        for j in (parsed[i]):
+            res = j
+
+    # Variables for testing
+    bookisbn = book.isbn
+
+    # Get all the reviews for the given book.
+    allreviews = review.query.filter_by(isbn=bookisbn).all()
+    if request.method == "POST":
```

Comment:
A user should be able to post only a single review on a particular book, where are you validating that?

Reply:
Bro In 139 line I am querying the database to check the particular user has given reviewed or not